### PR TITLE
Implemented all required network subcommands

### PR DESCRIPTION
**Pull request by tboerger**
The code is currently not DRY and I have not integrated tests yet. Hopefully we can take some time afterwards to reduce the duplication and to make it really rocksolid with tests.

**This depends on https://github.com/crowbar/crowbar-core/pull/95**

### CODE REVIEW

File: lib/crowbar/client/request/network.rb [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            name,
+            service,
+            network,
+            range,
+            suggestion
+          )
+            result = self.class.post(
+              "/crowbar/network/1.0/allocate_virtual_ip/#{name}.json",
+              body: {
+                name: service,
```
Comment (vuntz) [inline lib/crowbar/client/request/network.rb]: I find it confusing that we have a `name` variable, but that the `service` variable is the one passed for the `name` argument.

File: lib/crowbar/client/command/network.rb [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+                      say "Successfully deallocated the IP"
+                    else
+                      exit_now! request.parsed_response["error"]
+                    end
+                  end
+                end
+              end
+            end
+
+            parent.desc "Actions for real IP addresses"
```
Comment (vuntz) [inline lib/crowbar/client/command/network.rb]: It's not `real`, it's IP addresses for nodes.

File: lib/crowbar/client/request/network.rb [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+#
+
+module Crowbar
+  module Client
+    class Request
+      module Network
+        extend ActiveSupport::Concern
+
+        included do
+          def network_allocate_virtual(
```
Comment (vuntz) [inline lib/crowbar/client/request/network.rb]: Should be named  `network_allocate_virtual_ip`; there could be other virtual things.

File: lib/crowbar/client/request/network.rb (hunk 1 above)
Comment (tboerger) [inline lib/crowbar/client/request/network.rb]: It's confusing but I wanted to avoid to rename the parameters, that way we are backward compatible.

**Comment by tboerger** [comment]
@vuntz fixed your comments.

File: lib/crowbar/client/request/network.rb (hunk 1 above)
Comment (vuntz) [inline lib/crowbar/client/request/network.rb]: My implicit request (sorry for not being explicit) was to rename one of the variable, which you did. Thanks!

**Comment by vuntz** [comment]
+1

**Comment by MaximilianMeister** [comment]
+1

**Pull request merged**